gridstatus/ieso.py

```diff
@@ -0,0 +1,302 @@
+import io
+import time
+import xml.etree.ElementTree as ET
+
+import pandas as pd
+import requests
+
+from gridstatus import utils
+from gridstatus.base import ISOBase, NotSupported
+from gridstatus.decorators import support_date_range
+from gridstatus.gs_logging import log
+
+FORECAST_DEMAND_INDEX_URL = "http://reports.ieso.ca/public/OntarioZonalDemand/"
+
+# Each forecast file contains data from the day in the filename and going forward
+# 34 days. The most recent file does not have a date in the filename.
+FORECASTED_DEMAND_TEMPLATE_URL = (
+    f"{FORECAST_DEMAND_INDEX_URL}/PUB_OntarioZonalDemand_YYYYMMDD.xml"
+)
+
+# The farthest in the past that forecast files are available
+MAXIMUM_DAYS_IN_PAST_FOR_FORECAST = 90
+# THE farthest in the future that forecasts are available. Note that there are not
+# files for these future forecasts, they are in the current day's file.
+MAXIMUM_DAYS_IN_FUTURE_FOR_FORECAST = 34
+
+# Actual demand goes back several decades, with each year in one file.
+EARLIEST_ACTUAL_DEMAND = "2002-05-01"
+ACTUAL_DEMAND_INDEX_URL = "http://reports.ieso.ca/public/DemandZonal/"
+
+# Specifying the year is enough because the yearly file is updated daily.
+ACTUAL_DEMAND_TEMPLATE_URL = f"{ACTUAL_DEMAND_INDEX_URL}/PUB_DemandZonal_YYYY.csv"
+
+INTERVAL_DURATION = pd.Timedelta(hours=1)
+
+
+class IESO(ISOBase):
+    """Independent Electricity System Operator (IESO)"""
+
+    name = "Independent Electricity System Operator"
+    iso_id = "ieso"
+
+    # All data is provided in EST, and does not change with DST. This means there are
+    # no repeated or missing hours in the raw data and we can safely use tz_localize
+    # without setting ambiguous or nonexistent times.
+    # https://www.ieso.ca/-/media/Files/IESO/Document-Library/engage/ca/ca-Introduction-to-the-Capacity-Auction.ashx
+    default_timezone = "EST"
+
+    status_homepage = "https://www.ieso.ca/en/Power-Data"
+
+    @support_date_range(frequency="YEAR_START")
+    def get_load(self, date, end=None, verbose=False):
+        """
+        Get hourly load by zone for a given date or from date to end date.
+
+        Args:
+            date (datetime.date): date to get load for. If end is None, returns
+                only data for this date.
+            end (datetime.date, optional): end date. Defaults to None. If provided,
+                returns data from date to end date.
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pd.DataFrame: zonal load as a wide table with columns for each zone
```

Comment:
I might have done this wrong because other `get_load` methods return a single "Load" column. In that case, I can reword this so this method is the total load and add another `get_load_by_zone` method to get the wide df.

Reply:
nah, this is approach is better. i believe in PJM and NYISO we return zonal loads

Reply:
i stand corrected. see comment below